openedx/features/course_duration_limits/access.py

```diff
@@ -112,28 +112,25 @@ def check_course_expired(user, course):
     return ACCESS_GRANTED
 
 
-def register_course_expired_message(request, course):
+def generate_course_expired_message(user, course):
     """
-    Add a banner notifying the user of the user course expiration date if it exists.
+    Generate the message for the user course expiration date if it exists.
     """
-    if not CourseDurationLimitConfig.enabled_for_enrollment(user=request.user, course_key=course.id):
+    if not CourseDurationLimitConfig.enabled_for_enrollment(user=user, course_key=course.id):
```

Comment:
what was the reason for the change from request.user to user?  
I think this could affect masquerade as when you're masquerading I think request.user is the staff user and user is the masquerade target  
If you could confirm that when masquerading as a specific username request.user and user are the same then this is fine